runtime/src/utxo.rs

```diff
@@ -101,11 +111,12 @@ decl_module! {
 
 		/// Handler called by the system on block finalization
 		fn on_finalize() {
```

Comment:
Are we sure that in the PoW implementation, `substrate service` or whichever service usually called this function, will still call this function? Since PoW blocks never finalize, doesn't this mean our rewards wont get distributed?

Reply:
Yes, this gets called from frame_executive [here](https://github.com/paritytech/substrate/blob/master/frame/executive/src/lib.rs#L264). The naming is a bit unfortunate because "finalize" in this context doesn't refer to the grandpa/deterministic-finality concept. Rather it refers to completing the execution of the block. Even in chains that do have Grandpa, this is called at the end of block execution, not when the block is grandpa-finalized.

You can confirm for yourself that this runs by starting the node with `./target/release/node-template --validator -lruntime=debug` and observing that these lines actually print to the terminal.

```
sp_runtime::print("transaction reward sent to");
sp_runtime::print(hash.as_fixed_bytes() as &[u8]);
```